### ISSUE TITLE
How to replace RoBERTa with BERT?

### ISSUE THREAD

**Issue by desmond1687** [ucasYW/Contextual-Interaction-for-AQA#3]
I tried replacing roberta-base with bert-base-uncased in run.sh, but encountered a bug when loading the tokenizer.
Did you use `bert-base-uncased`?
How can I correctly switch from RoBERTa to BERT?

Thanks a lot!

**Comment by ucasYW** [comment]
If you want to use bert instead of roberta. In tokenization_dagn.py, comment out line 37 and use code I comment in line 38, and same for the line 175 and 176. This issue occurs because there are some special tokens in roberta but no in bert.

**Comment by desmond1687** [comment]
@ucasYW Thank you for your response. 
I've adjusted the code as you said, but after running it, the results are poor with a Pearson correlation coefficient of only 0.3667. May I ask what you think the possible reasons could be?

**Comment by ucasYW** [comment]
did you see any warning or something, there are still some code about special token of  roberta.

**Comment by desmond1687** [comment]
No, I do not see any warning about the tokenizer.

**Comment by ucasYW** [comment]
I forgot about line 149 and 150. Comment line 149 and uncomment 150. This time should work.

**Comment by ucasYW** [comment]
Same in tokenization_dagn.py.

**Comment by desmond1687** [comment]
Thanks for your quick reply!
I think the bug is about the loading of the pre-trained BERT model.
Because I see the following warning after I change roberta to bert:

`
Some weights of DAGN were not initialized from the model check
point at bert-base-uncased and are newly initialized: ['bert.r
oberta.embeddings.word_embeddings.weight', 'bert.roberta.embed
dings.position_embeddings.weight', 'bert.roberta.embeddings.to
ken_type_embeddings.weight', 'bert.roberta.embeddings.LayerNor
m.weight', 'bert.roberta.embeddings.LayerNorm.bias', 'bert.rob
erta.encoder.layer.0.attention.self.query.weight', 'bert.rober
ta.encoder.layer.0.attention.self.query.bias', 'bert.roberta.e
ncoder.layer.0.attention.self.key.weight', 'bert.roberta.encod
er.layer.0.attention.self.key.bias', 'bert.roberta.encoder.lay
er.0.attention.self.value.weight', 'bert.roberta.encoder.layer
.0.attention.self.value.bias', 'bert.roberta.encoder.layer.0.a
ttention.output.dense.weight', 'bert.roberta.encoder.layer.0.a
ttention.output.dense.bias', 'bert.roberta.encoder.layer.0.att
ention.output.LayerNorm.weight', 'bert.roberta.encoder.layer.0
.attention.output.LayerNorm.bias', 'bert.roberta.encoder.layer
.0.intermediate.dense.weight', 'bert.roberta.encoder.layer.0.i
ntermediate.dense.bias', 'bert.roberta.encoder.layer.0.output.
dense.weight', 'bert.roberta.encoder.layer.0.output.dense.bias
', 'bert.roberta.encoder.layer.0.output.LayerNorm.weight', 'be
rt.roberta.encoder.layer.0.output.LayerNorm.bias' ....
`

Do you have any idea how to fix it?

**Comment by ucasYW** [comment]
Sorry, dude. I forgot the most important part in DAGN.py, comment line 65 and uncomment line 66.

**Comment by desmond1687** [comment]
Thanks again!
Now it works fine.
However, the results of BERT are still unsatisfactory:
    "pearsonr": 0.388,
    "spearmanr": 0.3552
Do you have any clues?

**Comment by ucasYW** [comment]
Maybe there are still something need to comment, I'll check later. Is it just bert without contrastive learning and dagn or with both.

**Comment by ucasYW** [comment]
Did you uncomment all the bert stuff in DAGN, line 319 and line 320?

**Comment by desmond1687** [comment]
> Did you uncomment all the bert stuff in DAGN, line 319 and line 320?

Yes, I did.

**Comment by desmond1687** [comment]
> Maybe there are still something need to comment, I'll check later. Is it just bert without contrastive learning and dagn or with both.

I only tried bert with both now.

**Comment by ucasYW** [comment]
I think it's because the dropout. Don't use dropout or make it smaller when you using bert.

**Comment by desmond1687** [comment]
You mean the `numnet_drop` or the `dropout_prob` in DAGN?
I tried to set `dropout_prob` to 0, but still poor performance.

**Comment by desmond1687** [comment]
Update, after removing the `dagn` I can get normal results on BERT. But the results are still poor with `dagn` on BERT.

**Comment by ucasYW** [comment]
There should be some bug with DAGN on bert, probably with the special token or the merge method. I'll check them and update later.
